server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributeHealth.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.awarenesshealth;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.Strings;
+import org.opensearch.common.collect.ImmutableOpenMap;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * Cluster Awareness health information
+ *
+ */
+public class ClusterAwarenessAttributeHealth implements Iterable<ClusterAwarenessAttributeValueHealth>, Writeable, ToXContentFragment {
+
+    private final String awarenessAttributeName;
+    private final List<ClusterAwarenessAttributeValueHealth> awarenessAttributeValueHealthList;
+
+    /**
+     * Creates Awareness AttributeValue Health information
+     *
+     * @param awarenessAttributeValue Awareness Attribute value ie zone, rack etc
+     * @param weightedRoutingWeights  Weights Map of Awareness Attribute
+     * @param forcedAwarenessSettings Forced zone awareness settings
+     * @param replicaEnforcement Replica Enforcement enabled or not
+     * @param shardAllocationPerNode Shard Allocation per node
+     * @param nodeMap The nodeMap stored in cluster state
+     * @param totalShards Total shards currently in cluster
+     */
+    public ClusterAwarenessAttributeHealth(
```

Comment:
You could use a BuilderPattern to build this

Reply:
The ClusterAwarenessAttributeValueHealth is immutable and needs previous value to calculate a new one and thus avoiding the builder pattern.